### ISSUE TITLE
External Links: Adding the option to customize which 4xx fail would be great

### ISSUE THREAD

**Issue by abethcrane** [gjtorikian/html-proofer#223]
Currently I'm seeing a lot of 429's to youtube, and I'd like to be able to ignore these.

**Comment by penibelst** [comment]
The existing option `href_ignore` should do the job.

**Comment by Andersos** [comment]
All my [Youtube links](https://travis-ci.org/finn-no/tech.finn.no/builds/70913887) are also returning 429. Would be nice to fix this in some other way than just ignoring the links.

**Comment by penibelst** [comment]
> Would be nice to fix this in some other way than just ignoring the links.

I have two different ways in mind.

1. Handle [429 status](https://tools.ietf.org/html/rfc6585) links as *undefined*, beside *valid* and *invalid*. They would appear in the output, but the run doesn’t fail.
2. Introduce a new option `url_status_ignore` as array of numbers.

**Comment by Andersos** [comment]
Has there ever been a consideration of dropping domains. I know you could write a regex but it seems more user friendly to just ignore an entire domain. I have primarily two domains I block and that is Linkedin because of the 999 status code and now Youtube because of the 429.

**Comment by penibelst** [comment]
> I know you could write a regex but it seems more user friendly to just ignore an entire domain.

I don’t think we want to introduce an extra option for domains only. A `domain_ignore` option doesn’t look more user friendly than `url_ignore` to me.

```ruby
# both options would have the same effect
:url_ignore => [/youtube\.com/, /linkedin\.com/]
:domain_ignore => ["youtube.com", "linkedin.com"]
```

**Comment by Andersos** [comment]
I didn't know there was a url_ignore and `url_ignore`looks just fine =)

**Comment by gary65710** [comment]
Going

**Comment by gary65710** [comment]
Cool

**Comment by gary65710** [comment]
Good

**Comment by gary65710** [comment]
Yeah

**Comment by gjtorikian** [comment]
Closed by HTMLProofer 3.0: https://github.com/gjtorikian/html-proofer/releases/tag/v3.0.0
